### ISSUE TITLE
Adaptive cards don't render the Image dimensions

### ISSUE THREAD

**Issue by harischristodoulidis** [microsoft/AdaptiveCards#3095]
I use this schema.d.ts file:

```
export declare type Size = 'auto' | 'stretch' | 'small' | 'medium' | 'large';
export declare type TextSize = 'small' | 'default' | 'medium' | 'large' | 'extraLarge';
export declare type HorizontalAlignment = 'left' | 'center' | 'right';
export declare type VerticalAlignment = 'top' | 'center' | 'bottom';
export declare type Spacing =
	| 'none'
	| 'small'
	| 'default'
	| 'medium'
	| 'large'
	| 'extraLarge'
	| 'padding';
export declare type TextWeight = 'lighter' | 'default' | 'bolder';
export declare type TextColor =
	| 'default'
	| 'dark'
	| 'light'
	| 'accent'
	| 'good'
	| 'warning'
	| 'attention';
export declare type ContainerStyle = 'default' | 'emphasis';
export declare type ImageStyle = 'default' | 'person';
export interface IAction {
	id: string;
	title?: string;
}
export interface ISubmitAction extends IAction {
	type: 'Action.Submit';
	data?: any;
}
export interface IOpenUrlAction extends IAction {
	type: 'Action.OpenUrl';
	url: string;
}
export interface IShowCardAction extends IAction {
	type: 'Action.ShowCard';
	card: IAdaptiveCard;
}
export interface ICardElement {
	id?: string;
	speak?: string;
	horizontalAlignment?: HorizontalAlignment;
	spacing?: Spacing;
	separator?: boolean;
	height?: 'auto' | 'stretch';
	[propName: string]: any;
}
export interface IBackgroundImage {
	url: string;
}
export interface ITextBlock extends ICardElement {
	type: 'TextBlock';
	size?: TextSize;
	weight?: TextWeight;
	color?: TextColor;
	text: string;
	isSubtle?: boolean;
	wrap?: boolean;
	maxLines?: number;
}
export interface IContainer extends ICardElement {
	type: 'Container';
	backgroundImage?: IBackgroundImage | string;
	style?: ContainerStyle;
	verticalContentAlignment?: VerticalAlignment;
	selectAction?: IAction;
	items?: ICardElement[];
}
export interface IColumn extends IContainer {
	width?: number | 'auto' | 'stretch' | 'auto';
}
export interface IColumnSet extends ICardElement {
	type: 'ColumnSet';
	columns: IColumn[];
}
export interface IFact {
	title: string;
	value: string;
	speak?: string;
}
export interface IFactSet extends ICardElement {
	type: 'FactSet';
	facts: IFact[];
}
export interface IImage extends ICardElement {
	type: 'Image';
	altText?: string;
	selectAction?: IAction;
	size?: Size;
	style?: ImageStyle;
	url: string;
}
export interface IImageSet extends ICardElement {
	type: 'ImageSet';
	images: IImage[];
	size?: Size;
}
export interface IInput extends ICardElement {
	id: string;
	value?: string;
}
export interface IDateInput extends IInput {
	type: 'Input.Date';
	min?: string;
	max?: string;
	placeholder?: string;
}
export interface ITimeInput extends IInput {
	type: 'Input.Time';
	min?: string;
	max?: string;
	placeholder?: string;
}
export interface INumberInput extends IInput {
	type: 'Input.Number';
	min?: number;
	max?: number;
	placeholder?: string;
}
export interface ITextInput extends IInput {
	type: 'Input.Text';
	isMultiline?: boolean;
	maxLength?: number;
	placeholder?: string;
}
export interface IToggleInput extends IInput {
	type: 'Input.Toggle';
	title: string;
	valueOn?: string;
	valueOff?: string;
}
export interface IChoice {
	title: string;
	value: string;
}
export interface IChoiceSetInput extends IInput {
	type: 'Input.ChoiceSet';
	isMultiSelect?: boolean;
	style?: 'expanded' | 'compact';
	placeholder?: string;
	choices: IChoice[];
}
export interface IVersion {
	major: number;
	minor: number;
}
export interface IAdaptiveCard extends ICardElement {
	type: 'AdaptiveCard';
	version?: IVersion | string;
	backgroundImage?: IBackgroundImage | string;
	body?: (ITextBlock | IImage | IImageSet | IFactSet | IColumnSet | IContainer)[];
	actions?: (ISubmitAction | IOpenUrlAction | IShowCardAction)[];
	speak?: string;
}
```
The IImage interface inherits the properties of ICardElement
```
export interface IImage extends ICardElement {
	type: 'Image';
	altText?: string;
	selectAction?: IAction;
	size?: Size;
	style?: ImageStyle;
	url: string;
}
```

```
export interface ICardElement {
	id?: string;
	speak?: string;
	horizontalAlignment?: HorizontalAlignment;
	spacing?: Spacing;
	separator?: boolean;
	height?: 'auto' | 'stretch';
	[propName: string]: any;
}
```
ICardElement has a height property which can be 'auto' or 'stretch'
It does not have a width property

e.g. I take a json which has a column with this Image item:
```
{
  type: 'Column',
  style: 'default',
  items: [
    {
      type: 'Image',
      horizontalAlignment: 'Right',
      url:
        'http://cdn.onlinewebfonts.com/svg/img_286083.png',
      size: 'Small',
      width: '16px',
      height: '16px',
      altText: '',
    },
  ],
  width: 'auto',
},
```

The width and height of the Images are not rendered. The image is displayed with the default dimensions.

Do you know what is going wrong?

**Comment by harischristodoulidis** [comment]
I use this schema.d.ts file:

```
export declare type Size = 'auto' | 'stretch' | 'small' | 'medium' | 'large';
export declare type TextSize = 'small' | 'default' | 'medium' | 'large' | 'extraLarge';
export declare type HorizontalAlignment = 'left' | 'center' | 'right';
export declare type VerticalAlignment = 'top' | 'center' | 'bottom';
export declare type Spacing =
	| 'none'
	| 'small'
	| 'default'
	| 'medium'
	| 'large'
	| 'extraLarge'
	| 'padding';
export declare type TextWeight = 'lighter' | 'default' | 'bolder';
export declare type TextColor =
	| 'default'
	| 'dark'
	| 'light'
	| 'accent'
	| 'good'
	| 'warning'
	| 'attention';
export declare type ContainerStyle = 'default' | 'emphasis';
export declare type ImageStyle = 'default' | 'person';
export interface IAction {
	id: string;
	title?: string;
}
export interface ISubmitAction extends IAction {
	type: 'Action.Submit';
	data?: any;
}
export interface IOpenUrlAction extends IAction {
	type: 'Action.OpenUrl';
	url: string;
}
export interface IShowCardAction extends IAction {
	type: 'Action.ShowCard';
	card: IAdaptiveCard;
}
export interface ICardElement {
	id?: string;
	speak?: string;
	horizontalAlignment?: HorizontalAlignment;
	spacing?: Spacing;
	separator?: boolean;
	height?: 'auto' | 'stretch';
	[propName: string]: any;
}
export interface IBackgroundImage {
	url: string;
}
export interface ITextBlock extends ICardElement {
	type: 'TextBlock';
	size?: TextSize;
	weight?: TextWeight;
	color?: TextColor;
	text: string;
	isSubtle?: boolean;
	wrap?: boolean;
	maxLines?: number;
}
export interface IContainer extends ICardElement {
	type: 'Container';
	backgroundImage?: IBackgroundImage | string;
	style?: ContainerStyle;
	verticalContentAlignment?: VerticalAlignment;
	selectAction?: IAction;
	items?: ICardElement[];
}
export interface IColumn extends IContainer {
	width?: number | 'auto' | 'stretch' | 'auto';
}
export interface IColumnSet extends ICardElement {
	type: 'ColumnSet';
	columns: IColumn[];
}
export interface IFact {
	title: string;
	value: string;
	speak?: string;
}
export interface IFactSet extends ICardElement {
	type: 'FactSet';
	facts: IFact[];
}
export interface IImage extends ICardElement {
	type: 'Image';
	altText?: string;
	selectAction?: IAction;
	size?: Size;
	style?: ImageStyle;
	url: string;
}
export interface IImageSet extends ICardElement {
	type: 'ImageSet';
	images: IImage[];
	size?: Size;
}
export interface IInput extends ICardElement {
	id: string;
	value?: string;
}
export interface IDateInput extends IInput {
	type: 'Input.Date';
	min?: string;
	max?: string;
	placeholder?: string;
}
export interface ITimeInput extends IInput {
	type: 'Input.Time';
	min?: string;
	max?: string;
	placeholder?: string;
}
export interface INumberInput extends IInput {
	type: 'Input.Number';
	min?: number;
	max?: number;
	placeholder?: string;
}
export interface ITextInput extends IInput {
	type: 'Input.Text';
	isMultiline?: boolean;
	maxLength?: number;
	placeholder?: string;
}
export interface IToggleInput extends IInput {
	type: 'Input.Toggle';
	title: string;
	valueOn?: string;
	valueOff?: string;
}
export interface IChoice {
	title: string;
	value: string;
}
export interface IChoiceSetInput extends IInput {
	type: 'Input.ChoiceSet';
	isMultiSelect?: boolean;
	style?: 'expanded' | 'compact';
	placeholder?: string;
	choices: IChoice[];
}
export interface IVersion {
	major: number;
	minor: number;
}
export interface IAdaptiveCard extends ICardElement {
	type: 'AdaptiveCard';
	version?: IVersion | string;
	backgroundImage?: IBackgroundImage | string;
	body?: (ITextBlock | IImage | IImageSet | IFactSet | IColumnSet | IContainer)[];
	actions?: (ISubmitAction | IOpenUrlAction | IShowCardAction)[];
	speak?: string;
}
```
The IImage interface inherits the properties of ICardElement
```
export interface IImage extends ICardElement {
	type: 'Image';
	altText?: string;
	selectAction?: IAction;
	size?: Size;
	style?: ImageStyle;
	url: string;
}
```

```
export interface ICardElement {
	id?: string;
	speak?: string;
	horizontalAlignment?: HorizontalAlignment;
	spacing?: Spacing;
	separator?: boolean;
	height?: 'auto' | 'stretch';
	[propName: string]: any;
}
```
ICardElement has a height property which can be 'auto' or 'stretch'
It does not have a width property

e.g. I take a json which has a column with this Image item:
```
{
  type: 'Column',
  style: 'default',
  items: [
    {
      type: 'Image',
      horizontalAlignment: 'Right',
      url:
        'http://cdn.onlinewebfonts.com/svg/img_286083.png',
      size: 'Small',
      width: '16px',
      height: '16px',
      altText: '',
    },
  ],
  width: 'auto',
},
```

The width and height of the Images are not rendered. The image is displayed with the default dimensions.

Do you know what is going wrong?

**Comment by andrewleader** [comment]
@dclaux do you know what's up here? This seems to be related to the `schema.d.ts` file, I looked at that yesterday and noticed it's missing a ton of properties, maybe it's some deprecated thing we don't use anymore? Can you help here?

**Comment by andrewleader** [comment]
Ping @dclaux

**Comment by dclaux** [comment]
@harischristodoulidis can I ask how/why you are using the schema? Is it to get code completion support as you write Adaptive Card JSON payloads? Something else?

I am asking because I would like to retire these schema interfaces which serve little purpose. If you are building Adaptive Cards dynamically in code, I recommend using the object model instead, e.g.:

```typescript
var myTextBlock = new TextBlock();
myTextBlock.text = "Hello world";

var myCard = new AdaptiveCard();
myCard.addItem(myTextBlock);

myCard.render();
```

**Comment by harischristodoulidis** [comment]
We use this schema given from MS teams and it is not allowed to change it.
Then we created a common file for Mapping the style we are defining and implementing.
e.g. about size:
`export declare type Size = 'auto' | 'stretch' | 'small' | 'medium' | 'large';`
I can map the small size to be 16px or anything I want.
But I cannot map the Image height and width because these properties are missing.

I created the same issue at this channel:
https://github.com/microsoft/BotFramework-WebChat/issues/2092 and they told me that
"Web Chat v4 is currently the only channel that supports the most recent release of AdaptiveCards, so if you try to set those properties in any other channel (Web Chat v3, Test in Web Chat in the Azure Portal, Teams, Skype, Messenger, etc.), they will be ignored by the client's card rendered".

Thanks in advance!
